### PR TITLE
Separate plusargs per memory channel and synthesized print settings in manager

### PR DESCRIPTION
**Pull request by zhemao**
Three unrelated changes here

1. Make dramsim a standalone makefrag variable so that it's easier to disable for midas-level sim
2. Allow the DRAM timing model for each channel to have a different set of parameters
3. Expose options for synthesized prints in the config_runtime.ini

### CODE REVIEW

File: sim/midas/src/main/scala/midas/models/dram/FASEDMemoryTimingModel.scala [hunk 1]
```diff
@@ -22,6 +22,10 @@ import Console.{UNDERLINED, RESET}
 
 import java.io.{File, FileWriter}
 
+object ModelID {
```
Comment (davidbiancolin) [inline sim/midas/src/main/scala/midas/models/dram/FASEDMemoryTimingModel.scala:25]: I mentioned this earlier but we should reuse the ID in Widget.scala. We could add a method, `getInstanceID` there.

File: sim/midas/src/main/scala/midas/models/dram/FASEDMemoryTimingModel.scala [hunk 2]
```diff
@@ -552,9 +552,9 @@ class FASEDMemoryTimingModel(completeConfig: CompleteConfig, hostParams: Paramet
     // Accepts an elaborated memory model and generates a runtime configuration for it
     private def emitSettings(fileName: String, settings: Seq[(String, String)])(implicit p: Parameters): Unit = {
       val file = new File(p(OutputDir), fileName)
-      val writer = new FileWriter(file)
+      val writer = new FileWriter(file, wId != 0)
```
Comment (davidbiancolin) [inline sim/midas/src/main/scala/midas/models/dram/FASEDMemoryTimingModel.scala:555]: i'm asusming this is to denote appending to an existing file --> can you include the kwarg if it has a good name

File: deploy/sample-backup-configs/sample_config_runtime.ini [hunk 3]
```diff
@@ -57,3 +57,8 @@ suffixtag=
 # In general, this is not required to produce deterministic simulations on
 # target machines running linux. Enable if you observe simulation non-determinism.
 zerooutdram=no
+
+[synthprint]
```
Comment (davidbiancolin) [inline deploy/sample-backup-configs/sample_config_runtime.ini:61]: This probably needs a comment explaining what a base cycle is, or a link to the docs.

File: sim/src/main/makefrag/firesim/Makefrag [hunk 4]
```diff
@@ -95,6 +95,7 @@ EXTRA_VERILATOR_FLAGS = \
 # SW RTL Simulation Args -- for MIDAS- & FPGA-level Simulation #
 ################################################################
 TIMEOUT_CYCLES = 100000000
+DRAMSIM = +dramsim
```
Comment (davidbiancolin) [inline sim/src/main/makefrag/firesim/Makefrag:98]: how about HOST_MEM_ARGS

File: sim/midas/src/main/scala/midas/models/dram/FASEDMemoryTimingModel.scala (hunk 2 above)
Comment (zhemao) [reply sim/midas/src/main/scala/midas/models/dram/FASEDMemoryTimingModel.scala:555]: This is a java class, so  I don't think keyword arguments work.

Comment (davidbiancolin) [reply sim/midas/src/main/scala/midas/models/dram/FASEDMemoryTimingModel.scala:555]: Huh, the more you know.

**Pull request merged**